File: app/src/main/java/com/sysu/edu/view/RecyclerViewHolder.java
package com.sysu.edu.view;

import androidx.annotation.NonNull;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

public class RecyclerViewHolder<T extends ViewDataBinding> extends RecyclerView.ViewHolder {
    
    public final T binding;
    
    public RecyclerViewHolder(@NonNull T binding) {
        super(binding.getRoot());
        this.binding = binding;
    }
}

File: app/src/main/java/com/miyuyan/sysuer/view/RecyclerViewHolder.kt
package com.miyuyan.sysuer.view

import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

open class RecyclerViewHolder<T : ViewDataBinding?>(binding: T) :
	RecyclerView.ViewHolder(binding!!.root) {
	val binding: T? = binding
}
